### PR TITLE
Fix H-label classification

### PR DESCRIPTION
**Pull request by sungmanc**
### Summary

Related ticket: [CVS-114893](https://jira.devtools.intel.com/browse/CVS-114893)

**Issue description**
When there are empty heads that include no annotation, it should be ignored at loss calculation.
However, since it also affects the loss calculation, the loss is going to NaN. 

i.e. figure below represents the 3rd head is an empty head.
image

In this case, we should ignore the empty head to avoid the NaN loss.
```python3
  valid_mask = head_gt >= 0         ### In the case of empty head, valid_mask is []
  head_gt = head_gt[valid_mask].long()     ###  So, head_gt = []
  head_logits = head_logits[valid_mask, :]
  multiclass_loss = self.loss(head_logits, head_gt) ### Therefore this loss is going to NaN
  losses["loss"] += multiclass_loss
```

**This PR includes**
* Add logic: check the empty heads that include no annotation.
* Add logic: If there are empty heads, it will be filtered out at loss calculation.
* Add logic: patching the hierarchical information for cfg.model
* Add unit tests to check the functionality of empty heads

### Checklist

- [X] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [X] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Comment by sungmanc** [comment]
Now, we could check the model metric when empty heads exist
image

['Dog', 'Cat', 'Person', 'Supported animal'] labels are not used in this project
image

File: src/otx/algorithms/classification/adapters/mmcls/datasets/otx_datasets.py [hunk 1]
```diff
@@ -329,6 +327,19 @@ def load_annotations(self):
...
+        """Update heads information to find the empty heads.
+
+        If there are no annotations at a specific head, this should be filtered out to calculate loss correctly.
+        """
+        num_cls_heads = self.hierarchical_info["num_multiclass_heads"]
+        for head_idx in range(num_cls_heads):
+            labels_in_head = self.gt_labels[:, head_idx]  # type: ignore[call-overload]
+            if max(labels_in_head) < 0:
+                self.hierarchical_info["empty_multiclass_head_indices"].append(head_idx)
+
```
Comment (goodsong81) [inline src/otx/algorithms/classification/adapters/mmcls/datasets/otx_datasets.py:342]: It's weird to see "head" information in "dataset". Let's change to other terms in the future. Maybe, "empty_multiclass_label_indices"?

**Review by goodsong81 — APPROVED** [review]
Thank you for this fix. Looks nicer than before.

File: src/otx/algorithms/classification/adapters/mmcls/datasets/otx_datasets.py (hunk 1 above)
Comment (sungmanc) [reply src/otx/algorithms/classification/adapters/mmcls/datasets/otx_datasets.py:342]: Agree, I think it could be done at the refactoring phase to unify Multi-class / Multi-label / H-label Classification

**Review by jaegukhyun — APPROVED** [review]
